numba/tests/test_codegen.py

```diff
@@ -213,9 +213,12 @@ def test_noinline_on_main_call(self):
         # Checks that the cpython and cfunc wrapper produces a call with the
         # "noinline" attr present for the decorated function.
 
+        # Creating a refcounted object induces a side-effect that prevents IPO
+        # from eliding the call to the function with LLVM 15. See Issue #9658:
+        # https://github.com/numba/numba/issues/9658
         @njit
         def foo():
-            pass
+            return list([1])
```

Comment:
Yes, that's what i think is needed

Reply:
Thanks - I aimed to keep the test close to its original intent with this change.